framework/moose.mk

```diff
@@ -219,10 +219,17 @@ clobberall:: clobber
 
 compile_commands_all_srcfiles := $(moose_srcfiles) $(srcfiles)
 compile_commands.json::
-	@echo $(CURDIR) > .compile_commands.json
-	@echo $(libmesh_CXX) >> .compile_commands.json
-	@echo $(libmesh_CPPFLAGS) $(libmesh_CXXFLAGS) $(app_INCLUDES) $(libmesh_INCLUDE) $(ADDITIONAL_INCLUDES) >> .compile_commands.json
-	@echo $(compile_commands_all_srcfiles) >> .compile_commands.json
+	ifeq (4.0,$(firstword $(sort $(MAKE_VERSION) 4.0)))
```

Comment:
I just discovered that there is whitespace before `ifeq`, `else`, and `endif` in this commit. As a result, Make incorrectly interprets these as bash directives and the whole process fails. What is the best way to *>ahem<* patch this patch?